llm_studio/src/datasets/text_causal_language_modeling_ds.py

```diff
@@ -303,94 +303,68 @@ def sanity_check(cls, df: pd.DataFrame, cfg: Any, mode: str = "train"):
 
     def __getitem__(self, idx: int) -> Dict:
         """Returns a single dataset item."""
+        return self._read_data(idx=self.indices[idx])
```

Comment:
what is the point here, to call another function. 
Move everything from `_read_data` to here?